objectstate.h

```diff
@@ -0,0 +1,90 @@
+#ifndef OBJECTSTATE_H
+#define OBJECTSTATE_H
+/*
+	Copyright 2012 Benjamin Vedder	benjamin@vedder.se
+			  2020 Marvin Damschen  marvin.damschen@ri.se
+			  2021 Lukas Wikander	lukas.wikander@astazero.com
+
+	This program is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+	*/
+
+
+#include <QObject>
+#include <QVector>
+#include <QVector3D>
```

Comment:
QVector3D lives in Qt's GUI library, but this class and its children need to work without it (see the '#ifdef QT_GUI_LIB' part)